server/src/main/java/com/coincoinche/engine/BiddingMove.java

```diff
@@ -66,6 +68,30 @@ public static BiddingMove surcoincheMove() {
     return new BiddingMove(null, Special.SURCOINCHE);
   }
 
+  /**
+   * Create a bidding move from a PlayerBade event.
+   *
+   * @param event - player bade event from the client.
+   * @return the newly constructed bidding move.
+   */
+  public static BiddingMove fromEvent(PlayerBadeEvent event) {
+    Special special = event.getSpecial();
+    if (special == null) {
+      return contractMove(Contract.pointsContract(event.getValue(), event.getSuit()));
```

Comment:
For client-server communication, they are handled via their value (that's how they are declared in the engine) for simplicity. 
On the client, I added an adapter function to get the displayed value (i.e. display 'CAPOT' instead of 250).

Reply:
I removed the check on the method valueOfByPoint to prevent getting capot or generale via their value, it simplifies everything because all the value contracts are handled the same way. The client just have to be aware that those values need to be displayed differently.